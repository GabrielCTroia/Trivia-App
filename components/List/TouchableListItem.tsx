import React, { FunctionComponent } from 'react';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { ListItem, ListItemProps } from './ListItem';
import { Ionicons } from '@expo/vector-icons';
import * as Colors from '../../styles/Colors';


export type TouchableListItemProps = ListItemProps & {
  onPress: () => void;
};

export const TouchableListItem: FunctionComponent<TouchableListItemProps> = (props) => {
  const icon = <Ionicons name="ios-arrow-forward" color={Colors.lightShade} size={22} />;

  return (
    <TouchableOpacity onPress={props.onPress}>
      <ListItem
        rightSide={icon}
        style={props.style}
        {...props}
      />
    </TouchableOpacity>
  );
}
