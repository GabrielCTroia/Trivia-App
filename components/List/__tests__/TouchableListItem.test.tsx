import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { TouchableListItem } from '../TouchableListItem';

// TODO Test Fails for weird reason. Need to investigate
xit('renders correctly', () => {
  const tree = renderer.create(<TouchableListItem content="test" onPress={() => null} />).toJSON();

  expect(tree).toMatchSnapshot();
});
