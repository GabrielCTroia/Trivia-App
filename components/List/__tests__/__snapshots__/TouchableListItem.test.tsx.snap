// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E0E0E0",
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingTop": 20,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#828282",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    test
  </Text>
</View>
`;
